.travis.yml

```diff
@@ -1,18 +1,47 @@
+---
+dist: xenial
 language: ruby
-
-rvm:
-  - 2.3.1
-  - 2.4.0
-
-install: bundle install --without="development system_tests"
+cache: bundler
+before_install:
+  - gem update --system
+  - gem update bundler
+  - bundle --version
 script:
-  - bundle exec rake
-env:
-  - PUPPET_VERSION=4.10
-  - PUPPET_VERSION=5.5
-
+  - 'bundle exec rake $CHECK'
+matrix:
+  fast_finish: true
+  include:
+  - rvm: 2.4.4
+    bundler_args: --without system_tests development release
+    env: PUPPET_VERSION="~> 5.0" CHECK=test
+  - rvm: 2.5.3
+    bundler_args: --without system_tests development release
+    env: PUPPET_VERSION="~> 6.0" CHECK=test_with_coveralls
+  - rvm: 2.5.3
+    bundler_args: --without system_tests development release
+    env: PUPPET_VERSION="~> 6.0" CHECK=rubocop
+  - rvm: 2.4.4
+    bundler_args: --without system_tests development release
+    env: PUPPET_VERSION="~> 5.0" CHECK=build DEPLOY_TO_FORGE=yes
 branches:
   only:
-    - master
-    - develop
-
+  - master
+  - /^v\d/
+notifications:
+  email: false
+  irc:
+    on_success: always
+    on_failure: always
+    channels:
+      - "chat.freenode.org#voxpupuli-notifications"
+deploy:
+  provider: puppetforge
+  user: puppet
+  password:
+    secure: ""
```

Comment:
You could also add the secret from plumbing to `.sync.yml` and regenerate this file.